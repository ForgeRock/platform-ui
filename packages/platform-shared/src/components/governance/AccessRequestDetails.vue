<!-- Copyright (c) 2024 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <BContainer class="my-5">
    <template v-if="!isLoading">
      <!-- Header -->
      <FrRequestHeader
        :item="item" />

      <!-- Actions -->
      <FrRequestActions
        v-if="isActive && type === detailTypes.ADMIN_REQUEST"
        :permissions="{ reassign: true }"
        :type="detailTypes.ADMIN_REQUEST"
        @action="openModal($event)"
        class="mb-4" />

      <!-- Request details -->
      <BCard
        data-testId="request-detail"
        class="mb-3"
        no-body>
        <FrRequestDetails
          @add-comment="openModal('COMMENT')"
          :hide-actions="{ modify: true }"
          :item="item" />
      </BCard>

      <!-- Cancel panel -->
      <BCard
        data-testId="request-detail-cancel"
        v-if="isActive"
        class="mb-5">
        <h3 class="h5 card-title">
          {{ $t('governance.requestModal.cancelRequest') }}
        </h3>
        <p class="text-danger">
          {{ $t('common.cannotBeUndone') }}
        </p>
        <BButton
          variant="danger"
          @click="openModal('CANCEL')">
          {{ $t('governance.requestModal.cancelRequest') }}
        </BButton>
      </BCard>
    </template>
    <FrRequestModal
      :type="modalType"
      :item="item"
      @modal-closed="modalType = null"
      @update-item="getRequestData"
      @update-list="$emit('navigate-to-list')" />
  </BContainer>
</template>

<script setup>
import { computed, ref } from 'vue';
import {
  BButton,
  BCard,
  BContainer,
} from 'bootstrap-vue';
import { showErrorMessage } from '@forgerock/platform-shared/src/utils/notification';
import useBvModal from '@forgerock/platform-shared/src/composables/bvModal';
import { getRequest } from '@forgerock/platform-shared/src/api/governance/AccessRequestApi';
import {
  detailTypes,
  getFormattedRequest,
  getRequestTypeDisplayName,
} from '@forgerock/platform-shared/src/utils/governance/AccessRequestUtils';
import { REQUEST_MODAL_TYPES } from '@forgerock/platform-shared/src/utils/governance/constants';
import FrRequestModal from '@forgerock/platform-shared/src/components/governance/RequestModal/RequestModal';
import FrRequestDetails from '@forgerock/platform-shared/src/components/governance/RequestDetails/RequestDetails';
import FrRequestHeader from '@forgerock/platform-shared/src/components/governance/RequestDetails/RequestHeader';
import FrRequestActions from '@forgerock/platform-shared/src/components/governance/RequestDetails/RequestActions';
import i18n from '@/i18n';

const props = defineProps({
  requestId: {
    type: String,
    required: true,
  },
  type: {
    type: String,
    default: '',
  },
});

defineEmits(['navigate-to-list']);

// Composables
const { bvModal } = useBvModal();

// Data
const item = ref({});
const isLoading = ref(true);
const modalType = ref('');

const isActive = computed(() => item.value.rawData?.decision?.status === 'in-progress');

async function getRequestData() {
  isLoading.value = true;
  try {
    const { data } = await getRequest(props.requestId);
    const { data: requestTypeData } = await getRequestTypeDisplayName(data.requestType);
    data.requestTypeDisplayName = requestTypeData.displayName;
    item.value = getFormattedRequest(data);
  } catch (error) {
    showErrorMessage(error, i18n.global.t('governance.accessRequest.myRequests.errorGettingRequests'));
  } finally {
    isLoading.value = false;
  }
}

function openModal(type) {
  modalType.value = REQUEST_MODAL_TYPES[type];
  bvModal.value.show('request_modal');
}

getRequestData();
</script>
