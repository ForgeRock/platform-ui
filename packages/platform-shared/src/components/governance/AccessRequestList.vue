<!-- Copyright (c) 2023-2025 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <div>
    <slot name="header" />
    <FrSpinner
      v-if="isLoading"
      class="py-5" />
    <BTable
      v-else-if="items.length"
      :aria-label="listName"
      class="border-top mb-0"
      hover
      tbody-tr-class="cursor-pointer"
      :fields="fields"
      :items="items"
      @row-clicked="$emit('open-detail', $event);">
      <template #cell(details)="{ item }">
        <BMedia
          v-if="item.details"
          data-testId="request-detail-header"
          class="align-items-center">
          <BMediaBody class="align-self-center text-truncate">
            <h2 class="h5 mb-2">
              {{ item.itemName }}
            </h2>
            <div class="mb-1">
              <span class="text-dark mr-1">
                {{ item.details.requestedBy }}
              </span>
              <span class="text-muted mr-1">
                {{ $t('governance.accessRequest.submittedARequest') }}
              </span>
              <template v-if="item.details.requestedFor">
                <span class="text-muted mr-1">
                  {{ $t('common.for').toLowerCase() }}
                </span>
                <span class="text-dark">
                  {{ item.details.requestedFor }}
                </span>
              </template>
            </div>
            <div>
              <small
                class="text-muted">
                {{ $t('governance.accessRequest.idLabel', { id: item.details.id }) }}
              </small>
              <BImg
                :alt="getPriorityImageAltText(item.details.priority)"
                class="ml-2"
                height="24"
                :src="getPriorityImageSrc(item.details.priority)" />
            </div>
          </BMediaBody>
        </BMedia>
      </template>
      <template #cell(date)="{ item }">
        <small class="text-muted">
          {{ item.details.date }}
        </small>
      </template>
      <template #cell(actions)="{ item }">
        <slot
          name="actions"
          :item="item" />
      </template>
    </BTable>
    <template v-else>
      <slot name="no-data" />
    </template>
    <slot name="footer" />
  </div>
</template>

<script setup>
/**
 * Displays the list of Access Request.
 * @component AccessRequestList
 * @prop {Boolean} isLoading - Determines if the information is loading
 * @prop {Array} requests - All users requests
 */
import {
  BImg,
  BTable,
  BMedia,
  BMediaBody,
} from 'bootstrap-vue';
import { ref, watch } from 'vue';
import FrSpinner from '@forgerock/platform-shared/src/components/Spinner';
import { buildRequestDisplay, getPriorityImageSrc, getPriorityImageAltText } from '@forgerock/platform-shared/src/utils/governance/AccessRequestUtils';
import i18n from '@/i18n';

const prop = defineProps({
  isLoading: {
    type: Boolean,
    default: false,
  },
  requests: {
    type: Array,
    default: () => [],
  },
  listName: {
    type: String,
    default: '',
  },
});

defineEmits(['open-detail']);

const items = ref([]);
const fields = [
  {
    key: 'details',
    label: i18n.global.t('governance.accessRequest.newRequest.request'),
    class: 'w-65',
  },
  {
    key: 'date',
    label: i18n.global.t('governance.accessRequest.requestDate'),
  },
  {
    key: 'actions',
    label: '',
  },
];

watch(() => prop.requests, (newRequests) => {
  items.value = buildRequestDisplay(newRequests).map((item) => ({
    ...item,
    itemName: item.details?.isCustom
      ? item.details.type
      : `${item.details.type}: ${item.details.name}`,
  }));
}, { immediate: true });

</script>
<style lang="scss" scoped>
.w-200px {
  width: 200px;
}

:deep(.w-65) {
  width: 65% !important;
}
</style>
