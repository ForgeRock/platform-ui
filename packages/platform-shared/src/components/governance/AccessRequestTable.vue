<!-- Copyright (c) 2024-2025 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <BCard no-body>
    <FrAccessRequestList
      :is-loading="isLoading"
      :list-name="title"
      :requests="accessRequests"
      @open-detail="$emit('navigate-to-details', $event)">
      <template #header>
        <FrRequestToolbar
          v-model:num-filters="numFilters"
          :sort-by-options="sortByOptions"
          :status-options="statusOptions"
          @filter-change="filterHandler({ filter: $event })"
          @sort-change="filterHandler({ sortKeys: $event })"
          @sort-direction-change="filterHandler({ sortDir: $event })"
          @status-change="filterHandler({ status: $event })" />
      </template>
      <template #no-data>
        <FrNoData
          class="mb-4 border-top"
          icon="person_add"
          :card="false"
          :subtitle="$t('governance.accessRequest.noRequests', { status: getStatusText(statusOptions, status) })" />
      </template>
      <template #actions="{ item }">
        <FrRequestActionsCell
          v-if="status === 'in-progress'"
          class="mr-3"
          :item="item"
          :type="allowForwarding ? detailTypes.ADMIN_REQUEST : detailTypes.USER_REQUEST"
          @action="handleAction($event, item)" />
      </template>
    </FrAccessRequestList>
    <FrPagination
      v-if="totalRows"
      v-model="currentPage"
      :per-page="pageSize"
      :total-rows="totalRows"
      @input="filterHandler({ currentPage: $event })"
      @on-page-size-change="filterHandler({ pageSize: $event })" />
  </BCard>
  <FrRequestModal
    :type="modalType"
    :item="modalItem"
    @modal-closed="modalItem = {}"
    @update-list="loadRequests(true)" />
</template>

<script setup>
import {
  BCard,
} from 'bootstrap-vue';
import { ref } from 'vue';
import useBvModal from '@forgerock/platform-shared/src/composables/bvModal';
import FrPagination from '@forgerock/platform-shared/src/components/Pagination';
import FrNoData from '@forgerock/platform-shared/src/components/NoData';
import {
  detailTypes,
  getRequestFilter,
  getStatusText,
  sortByOptions,
  sortKeysMap,
} from '@forgerock/platform-shared/src/utils/governance/AccessRequestUtils';
import { REQUEST_MODAL_TYPES } from '@forgerock/platform-shared/src/utils/governance/constants';
import FrAccessRequestList from '@forgerock/platform-shared/src/components/governance/AccessRequestList';
import FrRequestActionsCell from '@forgerock/platform-shared/src/components/governance/RequestDetails/RequestActionsCell';
import FrRequestToolbar from '@forgerock/platform-shared/src/components/governance/RequestToolbar';
import FrRequestModal from '@forgerock/platform-shared/src/components/governance/RequestModal/RequestModal';
import i18n from '@/i18n';

/**
 * Displays the list of Requests and the corresponding filters.
 * @component MyRequest
 */

const props = defineProps({
  accessRequests: {
    type: Array,
    default: () => [],
  },
  totalRows: {
    type: Number,
    default: 0,
  },
  isLoading: {
    type: Boolean,
    default: false,
  },
  allowForwarding: {
    type: Boolean,
    default: false,
  },
  title: {
    type: String,
    default: '',
  },
});

// Emits
const emit = defineEmits(['load-requests', 'navigate-to-details']);

// Composables
const { bvModal } = useBvModal();

// Data
const currentPage = ref(1);
const filter = ref({});
const modalItem = ref({});
const modalType = ref(REQUEST_MODAL_TYPES.CANCEL);
const numFilters = ref(0);
const pageSize = ref(10);
const sortDir = ref('desc');
const sortKeys = ref('date');
const status = ref('in-progress');
const statusOptions = ref([
  {
    text: i18n.global.t('governance.status.pending'),
    value: 'in-progress',
  },
  {
    text: i18n.global.t('governance.status.complete'),
    value: 'complete',
  },
  {
    text: i18n.global.t('governance.status.canceled'),
    value: 'cancelled',
  },
]);

const componentRefs = new Map([
  ['currentPage', currentPage],
  ['filter', filter],
  ['pageSize', pageSize],
  ['sortDir', sortDir],
  ['sortDir', sortDir],
  ['sortKeys', sortKeys],
  ['status', status],
  ['totalRows', props.totalRows],
]);

/**
 * Get current users access requests based on query params and target filter
 */
async function loadRequests(goToFirstPage) {
  if (goToFirstPage) currentPage.value = 1;
  const payload = getRequestFilter(filter.value, status.value);
  const params = {
    pagedResultsOffset: (currentPage.value - 1) * pageSize.value,
    pageSize: pageSize.value,
    sortKeys: sortKeysMap[sortKeys.value],
    sortDir: sortDir.value,
  };

  if (sortKeys.value === 'date') params.sortType = 'date';
  emit('load-requests', params, payload);
}

/**
 * Opens a modal based on the provided item and type.
 * @param {Object} item - The item to show in the modal.
 * @param {string} type - The type of modal to open.
 */
function openModal(item, type) {
  modalItem.value = item;
  modalType.value = REQUEST_MODAL_TYPES[type];
  bvModal.value.show('request_modal');
}

/**
 * Handles the specified action for a given item.
 * @param {string} action - The action to be performed.
 * @param {Object} item - The item on which the action is to be performed.
 */
function handleAction(action, item) {
  if (action === 'DETAILS') emit('navigate-to-details', item);
  else openModal(item, action);
}

/**
 * Handles filtering requests as well as updates to pagination
 * @param {Object} property updated property
 */
function filterHandler(property) {
  const [key, value] = Object.entries(property)[0];
  componentRefs.get(key).value = value;
  const resetPaging = (key !== 'currentPage');
  loadRequests(resetPaging);
}
</script>
