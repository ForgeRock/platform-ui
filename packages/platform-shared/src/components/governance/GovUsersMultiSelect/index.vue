<!-- Copyright (c) 2023 ForgeRock. All rights reserved.

This software may be modified and distributed under the terms
of the MIT license. See the LICENSE file for details. -->
<template>
  <FrField
    v-model="selectValue"
    :description="description"
    :internal-search="false"
    :label="label"
    :options="options"
    :value="selectValue"
    @input="handleInput"
    @search-change="debouncedSearch"
    open-direction="bottom"
    type="multiselect">
    <template #tag="{ option, remove }">
      <span class="multiselect__tag">
        <BMedia no-body>
          <BImg
            height="24"
            width="24"
            class="mr-2 align-self-center"
            :src="option.value.profileImage || require('@forgerock/platform-shared/src/assets/images/avatar.png')" />
          <BMediaBody class="pl-1">
            <div
              class="mb-1 text-dark"
              tabindex="0">
              {{ option.value.name }}
            </div>
            <small class="text-muted">
              {{ option.value.userName }}
            </small>
          </BMediaBody>
          <span
            class="multiselect__tag-icon"
            tabindex="0"
            :aria-label="$t('common.remove')"
            @click.prevent="remove(option)"
            @keydown.enter="remove(option)" />
        </BMedia>
      </span>
    </template>
    <template #option="{ option }">
      <BMedia no-body>
        <BImg
          height="24"
          width="24"
          class="mr-2 align-self-center"
          :src="option.value.profileImage || require('@forgerock/platform-shared/src/assets/images/avatar.png')" />
        <BMediaBody class="pl-1">
          <div class="mb-1 text-dark">
            {{ option.value.name }}
          </div>
          <small class="text-muted">
            {{ option.value.userName }}
          </small>
        </BMediaBody>
      </BMedia>
    </template>
  </FrField>
</template>

<script>
import {
  debounce,
  filter,
  some,
} from 'lodash';
import { BImg, BMedia, BMediaBody } from 'bootstrap-vue';
import { getResource } from '@forgerock/platform-shared/src/api/governance/CommonsApi';
import FrField from '@forgerock/platform-shared/src/components/Field';
import NotificationMixin from '@forgerock/platform-shared/src/mixins/NotificationMixin/';

export default {
  name: 'GovUsersMultiSelect',
  components: {
    BImg,
    BMedia,
    BMediaBody,
    FrField,
  },
  mixins: [
    NotificationMixin,
  ],
  props: {
    description: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      debouncedSearch: debounce(this.search, 500),
      isSearching: false,
      options: [],
      selectValue: [],
    };
  },
  mounted() {
    this.getResourceList();
  },
  methods: {
    search(query) {
      this.getResourceList(query);
      this.isSearching = true;
    },
    getResourceList(queryString) {
      const pageSize = this.selectValue.length + 10;
      getResource('user', { queryString, pageSize })
        .then(({ data }) => {
          const parsedOptions = data.result.map((element) => ({
            value: {
              name: this.$t('common.userFullName', {
                givenName: element.givenName,
                sn: element.sn,
              }),
              id: element.id,
              profileImage: element.profileImage,
              userName: element.userName,
            },
          }));
          this.filterOptions(parsedOptions);
        })
        .catch((error) => {
          this.showErrorMessage(error, this.$t('governance.errors.getUsers'));
        });
    },
    filterOptions(parsedOptions) {
      const filteredOptions = filter(parsedOptions, ({ value }) => !some(this.selectValue, value));
      this.options = filteredOptions;
    },
    handleInput(event) {
      this.isSearching = false;
      this.$emit('input', event);
    },
  },
};
</script>
