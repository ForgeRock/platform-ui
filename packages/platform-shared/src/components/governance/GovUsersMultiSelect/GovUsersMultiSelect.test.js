/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { mount } from '@vue/test-utils';
import flushPromises from 'flush-promises';
import * as CommonsApi from '@forgerock/platform-shared/src/api/governance/CommonsApi';
import i18n from '@/i18n';
import GovUsersMultiSelect from './index';

const mountComponent = () => mount(GovUsersMultiSelect, {
  i18n,
  mocks: {
    $store: {
      state: {
        SharedStore: {
          uiConfig: {},
        },
      },
    },
  },
});

const usersOptions = jest.fn().mockReturnValue(Promise.resolve({
  data: {
    result: [
      {
        givenName: 'test1',
        id: 'userId1',
        profileImage: 'urlTest1',
        sn: 'user',
        userName: 'userName1',
      },
      {
        givenName: 'test2',
        id: 'userId2',
        profileImage: 'urlTest2',
        sn: 'user',
        userName: 'userName2',
      },
      {
        givenName: 'test3',
        id: 'userId3',
        profileImage: 'urlTest3',
        sn: 'user',
        userName: 'userName3',
      },
    ],
  },
}));

describe('GovUsersMultiSelect', () => {
  CommonsApi.getResource = usersOptions;

  it('shows all users on initial component loading', async () => {
    const wrapper = mountComponent();
    await flushPromises();

    expect(wrapper.vm.options).toEqual([
      {
        value: {
          name: 'test1 user',
          id: 'userId1',
          profileImage: 'urlTest1',
          userName: 'userName1',
        },
      },
      {
        value: {
          name: 'test2 user',
          id: 'userId2',
          profileImage: 'urlTest2',
          userName: 'userName2',
        },
      },
      {
        value: {
          name: 'test3 user',
          id: 'userId3',
          profileImage: 'urlTest3',
          userName: 'userName3',
        },
      },
    ]);
  });

  it('does not duplicate the selected option', async () => {
    const wrapper = mountComponent();
    await flushPromises();
    wrapper.vm.selectValue = [
      {
        name: 'test1 user',
        id: 'userId1',
        profileImage: 'urlTest1',
        userName: 'userName1',
      },
      {
        name: 'test2 user',
        id: 'userId2',
        profileImage: 'urlTest2',
        userName: 'userName2',
      },
    ];
    await flushPromises();
    wrapper.vm.filterOptions(wrapper.vm.options);

    expect(wrapper.vm.options).toEqual([
      {
        value: {
          name: 'test3 user',
          id: 'userId3',
          profileImage: 'urlTest3',
          userName: 'userName3',
        },
      },
    ]);
  });
});
