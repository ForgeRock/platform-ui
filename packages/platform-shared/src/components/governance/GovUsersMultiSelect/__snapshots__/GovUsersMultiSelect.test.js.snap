// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovUsersMultiSelect shows all users on initial component loading: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <FrField",
    "modelValue=",
    [],
    "onUpdate:modelValue=fn",
    "description=""",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (2)",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (3)",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (4)",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "appear" with v-bind value \`false\` will render appear="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "persisted" with v-bind value \`false\` will render persisted="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_ENUMERATED_COERCION) Enumerated attribute "spellcheck" with v-bind value \`false\` will render the value as-is instead of coercing the value to "true" in Vue 3. Always use explicit "true" or "false" values for enumerated attributes. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_ENUMERATED_COERCION: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "internal-search" with v-bind value \`false\` will render internal-search="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) (2)",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <Transition",
    "name="multiselect"",
    "persisted=""",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) Functional component <BMedia> should be defined as a plain function in Vue 3. The "functional" option has been removed. NOTE: Before migrating to use plain functions for functional components, first make sure that all async components usage have been migrated and its compat behavior has been disabled.
  Details: https://v3-migration.vuejs.org/breaking-changes/functional-components.html",
    "
",
    " at <Transition",
    "name="multiselect"",
    "persisted=""",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (2)",
    "
",
    " at <BMedia",
    "no-body=""",
    ">",
    "
",
    " at <Transition",
    "name="multiselect"",
    "persisted=""",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (3)",
    "
",
    " at <BImg",
    "height="24"",
    "width="24"",
    "class="mr-2 align-self-center"",
    " ...",
    ">",
    "
",
    " at <BMedia",
    "no-body=""",
    ">",
    "
",
    " at <Transition",
    "name="multiselect"",
    "persisted=""",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (4)",
    "
",
    " at <BMediaBody",
    "class="pl-1"",
    ">",
    "
",
    " at <BMedia",
    "no-body=""",
    ">",
    "
",
    " at <Transition",
    "name="multiselect"",
    "persisted=""",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "internal-search=false",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <FrField",
    "description=""",
    "internal-search=false",
    "label=""",
    " ...",
    ">",
    "
",
    " at <GovUsersMultiSelect",
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
